File: ImgManager/ImgManager/ts/contracts/SortingContract.ts
class SortingContract {
    page: number;
    pageSize: number;
    property: string;
    order: Order;

    public constructor(
        fields?: {
            page?: number,
            pageSize?: number,
            property?: string,
            order?: Order
        }) {
        if (fields) Object.assign(this, fields);
    }
}

enum Order {
    None = 0,
    Ascending,
    Descending
}

File: ImgManager/ImgManager/contracts/SortingContract.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Linq.Expressions;
using System.Web;

namespace ImgManager.contracts
{
    public struct SortingContract
    {
        public int page { get; set; }
        public int pageSize { get; set; }
        public string property { get; set; }
        public Order order { get; set; }
    }

    public enum Order
    {
        None = 0,
        Ascending,
        Descending
    }
}